imap_data_access/cli.py

```diff
@@ -99,13 +104,32 @@ def _query_parser(args: argparse.Namespace):
         "version",
         "extension",
     ]
+    latest_version = ""
     query_params = {
         key: value
         for key, value in vars(args).items()
         if key in valid_args and value is not None
     }
+
+    # removing version from query if it is 'latest', updating flag
+    if args.version == "latest":
+        del query_params["version"]
+        latest_version = "latest"
+
     query_results = imap_data_access.query(**query_params)
 
+    # if latest version was included in search then filter returned query for largest.
+    if latest_version == "latest":
+        max_version = max(
+            extract_version_number(each_dict.get("version"))
+            for each_dict in query_results
+        )
+        query_results = [
+            each_dict
+            for each_dict in query_results
+            if extract_version_number(each_dict["version"]) >= max_version
```

Comment:
```suggestion
            if extract_version_number(each_dict["version"]) == max_version
```

I think logically this should be an exact match?

Reply:
I agree with @bourque because the `max_version` is first assigned based on the max version found in the retrieved query. There is no way for any versions in the retrieved query to be greater than the `max_value`.